tests/test_socketdelay.js

```diff
@@ -43,59 +42,50 @@ test('calling socketDelay will emit a timeout', t => {
     .socketDelay(10000)
     .reply(200, 'OK')
 
-  let timedout = false
-  let ended = false
+  const onEnd = sinon.spy()
 
-  const req = http.request('http://example.test', function(res) {
+  const req = http.request('http://example.test', res => {
     res.setEncoding('utf8')
-
-    res.once('end', function() {
-      ended = true
-      if (!timedout) {
-        t.fail('socket did not timeout when idle')
-        t.end()
-      }
-    })
+    res.once('end', onEnd)
   })
 
-  req.setTimeout(5000, function() {
-    timedout = true
-    if (!ended) {
-      t.ok(true)
-      t.end()
-    }
+  req.setTimeout(5000, () => {
+    expect(onEnd).not.to.have.been.called()
+    t.end()
```

Comment:
Having two `t.end()` calls is confusing and not helpful (except in the case of test failures, though I don't think it is worth crafting our tests around failures, especially now that we're switching to Mocha which has reasonable timeouts and clear error messages).

These assertions also seemed convoluted and much more complicated than necessary.

What we want to have happen is that we wait 5 seconds and then pass the test, so long as the `'end'` event hasn't fired during that time. It's sufficient to simply assert that.